### PR TITLE
feat: created a user management controller for the admin to access

### PR DESCRIPTION
**Pull request by Hridansh70**
Introduced an AdminUserManagementController that can be accessed by any admin. This allows the admin to create various kinds of user accounts as well as delete them.
This was done to resolve the issue created: **abstract admin user management into a separate controller **.

### CODE REVIEW

File: src/controllers/AdminUserManagementController.java [hunk 1]
```diff
@@ -91,6 +91,14 @@ private void DisplaySuccessOnCreateAccount(UserData<?> user) {
...
-
+    private boolean DeleteUserHelper(String username) {
+        if (patientManager.deleteUser(username)) {
+            return true;
+        } else if (doctorManager.deleteUser(username)) {
+            return true;
+        } else if (secretaryManager.deleteUser(username)) {
+            return true;
+        } else return adminManager.deleteUser(username);
+    }
```
Comment (DanielDervishi) [inline src/controllers/AdminUserManagementController.java]: This method can be reduced using boolean logic.

File: src/controllers/AdminController.java [hunk 2]
```diff
@@ -46,94 +46,100 @@ public AdminController(Context context, AdminData adminData) {
...
+//        commands.put("create secretary", CreateSecretary());
+//        commands.put("create doctor", CreateDoctor());
+//        commands.put("create patient", CreatePatient());
         commands.put("change clinic info", changeClinicInformation());
         commands.put("change user password", changeUserPassword());
-        commands.put("delete user", deleteUser());
+//        commands.put("delete user", deleteUser());
         commands.put("delete self", deleteSelf());
-
+        commands.put("load user management system", LoadUserManagement());
```
Comment (DanielDervishi) [inline src/controllers/AdminController.java]: Could we change this to something easier to type like "manage users".

File: src/controllers/AdminController.java [hunk 3]
```diff
@@ -46,92 +45,24 @@ public AdminController(Context context, AdminData adminData) {
     @Override
     public LinkedHashMap<String, Command> AllCommands() {
         LinkedHashMap<String, Command> commands = new LinkedHashMap<>();
-        commands.put("create admin", CreateAdmin());
-        commands.put("create secretary", CreateSecretary());
-        commands.put("create doctor", CreateDoctor());
-        commands.put("create patient", CreatePatient());
         commands.put("change clinic info", changeClinicInformation());
         commands.put("change user password", changeUserPassword());
```
Comment (DanielDervishi) [inline src/controllers/AdminController.java]: Change user password should be moved to AdminUserManagementController.

File: src/controllers/AdminController.java [hunk 4]
```diff
@@ -46,92 +45,24 @@ public AdminController(Context context, AdminData adminData) {
...
-        commands.put("delete user", deleteUser());
         commands.put("delete self", deleteSelf());
-
+        commands.put("load user management system", LoadUserManagement());
```
Comment (DanielDervishi) [inline src/controllers/AdminController.java]: The ordering of the commands here is a bit odd. I would suggest starting with the commands that relate to the responsibility of the admin, such as managing users, changing clinic info and then delete self.

File: src/controllers/AdminUserManagementController.java (hunk 1 above)
Comment (ahmedkhalf) [reply src/controllers/AdminUserManagementController.java]:
```
return blabla |
        bloblo |
        bleble;
```

**Review by ahmedkhalf — APPROVED** [review]
Looks good.

**Review by DanielDervishi — APPROVED** [review]
Looks good to me.

**Pull request merged**